File: app/lib/data/model/category.dart
class Category {
  final BigInt id;
  final String name;
  final int numberOfThings;
  final String iconUrl;

  @override
  bool operator ==(Object other) =>
      identical(this, other) ||
      other is Category && runtimeType == other.runtimeType && id == other.id;

  @override
  int get hashCode => id.hashCode;

  const Category({
    required this.id,
    required this.name,
    required this.numberOfThings,
    required this.iconUrl,
  });

  Map<String, dynamic> toMap() {
    return {
      'id': id,
      'name': name,
      'number_of_things': numberOfThings,
      'icon_url': iconUrl,
    };
  }

  factory Category.fromMap(Map<String, dynamic> map) {
    return Category(
      id: BigInt.from(map['id']),
      name: map['name'] as String,
      numberOfThings: map['number_of_things'] as int,
      iconUrl: map['icon_url'] as String,
    );
  }
}

File: server/internal/model/category.go
package model

type Category struct {
	Id             int64  `json:"id,omitempty"`
	Name           string `json:"name,omitempty"`
	IconUrl        string `json:"icon_url,omitempty"`
	NumberOfThings int    `json:"number_of_things,omitempty"`
}

func MakeCategory(id int64, name string, iconUrl string, numberOfThings int) Category {
	return Category{id, name, iconUrl, numberOfThings}
}
